### PR TITLE
Repair auto detect and fetch upstream scripts

### PR DESCRIPTION
**Pull request by henrykironde**
Repair auto detect
    
Include the size of the type where needed.
Use length of encoded string to account for encoded chars
Do not hard code total number of scripts

### CODE REVIEW

File: retriever/lib/create_scripts.py [hunk 1]
```diff
@@ -90,6 +105,7 @@ def create_script_dict(allpacks, path, file, skip_lines):
     allpacks["citation"] = "FILL"
     allpacks["licenses"] = [{"name": "FILL"}]
     allpacks["keywords"] = []
+    allpacks["archived"] = "fill or remove this field if not archived"
```
Comment (ethanwhite) [inline retriever/lib/create_scripts.py]: I wouldn't be immediately clear on what this field should be. Is it a logical? If so what about "FILL: 'True' if data file is compressed and 'False' if it is not"?

Comment (ethanwhite) [reply retriever/lib/create_scripts.py]: Alternatively we could set it to False which will be the most common case and then include why/how to switch it in the docs.

**Comment by ethanwhite** [comment]
Looks great. One minor comment for discussion.

File: retriever/lib/create_scripts.py (hunk 1 above)
Comment (henrykironde) [reply retriever/lib/create_scripts.py]: This is only used when the dataset is stored as an archive. It takes on the value of the type of archive. ie, `zip`, `gz`, etc. I think we have this in docs somewhere. We can discus this later for more input. Thanks Ethan.

Comment (ethanwhite) [reply retriever/lib/create_scripts.py]: Sounds good. I'll go ahead and merge this in and we can modify that language later if we come up with something better.

**Pull request merged**